app/client/utils.js

```diff
@@ -306,6 +319,158 @@ export const processFiltersForQuery = filters => {
   return encoded_filters;
 };
 
+export const parseEncodedParams = (allowedFilters, params) => {
```

Comment:
So i could be understanding this wrong, but what I don't love here is that it seems like inputting allowedFilters is being heavily relied upon to do the parsing, when in theory there should be everything we need to parse correctly in the param string.
I'm my mind this increases the complexity of the parser significantly, and makes it less flexible, since we always need to have an opinion on allowedFilters.

The parser becomes pretty small (this is just a direct transposition of the python-side version that I threw together):
```
const parseEncodedParams = (params) => {
  let inequalityMapper = {
    'EQ': '=',
    'LT': '<',
    'GT': '>',
  };

  return params
    .split(":")
    .filter(item => item)
    .map((param, i) => {
      //Tokenize across parentheses
      let [attribute, comparator, value] = param.split(/[()]+/).filter(e => e);

      if (comparator === 'IN') {
        return {
          type: 'discrete',
          attribute: attribute,
          allowedValues: value.split(','),
          id: i+1
        }
      }

      return {
        type: inequalityMapper[comparator] || inequalityMapper['EQ'],
        attribute: attribute,
        threshold: value,
        id: i+1
      }
  })
};
```

I think passing in allowedFilters  does have some merit in using it to validate the param string (ie throwing an error if there's an `EQ` where there's supposed to be an `IN`), but even then, I'd think I think i'd be simpler to write that over two seperate functions, which each one having a specific task: one to parse into a dict, and then one to validate whether the filter is actually valid or not. Right now we just get TypeErrors and the like, when it'd be pretty simple to validate properly.

Reply:
Edit:

Errrgg just realised that `IN` can also be a boolean mapping (though they're treated the same everyhwere). That's annoying. Adding this still is a lot easier to understand as an overall function - no tripple-nested `if` statements (though just my opinion!)

```
     let matchedFilter = allowedFilters[attribute];
      if (!matchedFilter) {
        throw `Allowed Filter not found for attribute ${attribute}`
      }

      if (comparator === 'IN') {
        return {
          type: matchedFilter.type,
          attribute: attribute,
          allowedValues: value.split(','),
          id: i+1
        }
      }
```